indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -198,6 +197,30 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLock)
```

Comment:
Do you mean it should throw the exception when the `taskLock` is already in upgraded state (i.e. non-preemptible) and the upgraded `taskLock` is not present in `taskLocks`. Thus, the callers should provide `taskLock` which is preemptible (not-upgraded) or non-preemptible (upgraded) but already present in `taskLocks`